### DISCUSSION TITLE
How to do when not upgrade IN TIME?

### DISCUSSION THREAD

**Discussion by seantw1974** [filecoin-project/lotus#8980] [General Q&A] [answered]
1. not upgrade in time
2. lots of sectors in "CommitWait" or "PreCommitWait"
3. Some Lotus mpool pending --local
4. Nonce number is wrong ....
Please help how to correct ..

**Comment by magik6k** [comment, marked as answer]
For fixing bad nonces we have a [tool in lotus-shed](https://github.com/filecoin-project/lotus/blob/master/cmd/lotus-shed/nonce-fix.go) which will generate messages 'filling' gaps in nonces

`./lotus-shed noncefix --auto --addr f3...`, or if you know the bad nonces, `./lotus-shed noncefix --addr f3... --start [startnonce] --end [endnonce]`

**Comment by maweis1981** [comment]
A full workday to fix the same problem, I forgot to upgrade.
